client/src/pages/Feed.js

```diff
@@ -466,15 +564,44 @@ const Feed = (props) => {
       }
     };
 
+    const searchURL = () => {
+      if (searchKeyword)
+        return `&keywords=${encodeURIComponent(searchKeyword)}`;
+      else return "";
+    };
+
     const limit = PAGINATION_LIMIT;
     const skip = page * limit;
-    const baseURL = `/api/posts?&includeMeta=true&limit=${limit}&skip=${skip}`;
-    let endpoint = `${baseURL}${objectiveURL()}${filterURL()}`;
+    let baseURL = `/api/posts?includeMeta=true&limit=${limit}&skip=${skip}`;
+    switch (searchCategory) {
+      case "POSTS":
+        break;
+      case "INDIVIDUALS":
+        baseURL = `/api/users?includeMeta=true&limit=${limit}&skip=${skip}`;
+        break;
+      case "ORGANISATIONS":
+        baseURL = `/api/organisations/search?includeMeta=true&limit=${limit}&skip=${skip}`;
+        break;
+      default:
+        break;
+    }
+
+    let endpoint = `${baseURL}${objectiveURL()}${filterURL()}${searchURL()}`;
     postsDispatch({ type: FETCH_POSTS });
+
     try {
       const {
         data: { data: posts, meta },
       } = await axios.get(endpoint);
+      if (searchKeyword && window.dataLayer) {
+        // trigger custom GTM event
+        window.dataLayer.push({
```

Comment:
```suggestion
        TagManager.dataLayer({
          dataLayer: {
            event: "SEARCH_KEYWORD",
            keyword: searchKeyword,
            category: searchCategory || "POSTS",
            resultsCount: meta.total,
          }
        });

```

A bit nit-picky but can we use the gtm library for consistency?

You just need to also at the top:

```
import TagManager from "react-gtm-module";
```

Other than that I'm good code wise we can merge after deploy. @idjevm

Reply:
i am changing that, I will commit in few minutes